pkg/tools/kubectl_filter.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tools
+
+import (
+	"strings"
+)
+
+// KubectlModifiesResource determines if a kubectl command modifies resources.
+// Returns:
+// - "yes" if the command definitely modifies resources
+// - "no" if the command definitely doesn't modify resources
+// - "unknown" if we can't determine for sure
+func KubectlModifiesResource(command string) string {
+	// Skip commands that aren't kubectl commands
+	if !strings.Contains(command, "kubectl") {
+		return "unknown"
+	}
+
+	// Check for dry-run flag first - this overrides any other command type
+	if hasDryRunFlag(command) {
+		return "no"
+	}
+
+	// List of prefixes/keywords that indicate read-only operations
+	readOnlyOperations := []string{
+		"get", "describe", "explain", "top", "logs", "api-resources",
+		"api-versions", "version", "config view", "cluster-info",
+		"wait", "auth can-i", "diff", "kustomize", "help",
+		"options", "plugin", "proxy", "completion",
+		"alpha list", "alpha status", "alpha events",
+	}
+
+	// List of prefixes/keywords that indicate resource modification
+	modifyingOperations := []string{
+		"create", "apply", "edit", "delete", "patch", "replace",
+		"scale", "autoscale", "expose", "rollout", "run", "set",
+		"label", "annotate", "taint", "drain", "cordon", "uncordon",
+		"certificate approve", "certificate deny", "debug", "attach",
+		"cp", "exec", "port-forward", "auth reconcile",
+		"alpha events trigger",
+	}
```

Comment:
I am curious how did you construct these lists ? (is there a way to ensure that this list is exhaustive) ?